flax/nn/base.py

```diff
@@ -943,7 +945,13 @@ def store(self, value):
     frame = _top_frame('store')
     if not self._mutable:
       raise ValueError('Collection is not mutable. Use the `mutate` method to'
-                       'create a mutable copy.')
+                       ' create a mutable copy.')
+    master = utils._tracer_of_value(value)
+    value_level = master.level if master else -1000
```

Comment:
The magic constant bugs me but maybe it's fine, consider a short comment, e.g. "If you're inside a 1000-layer JAX transformation, all bets are off"